blueocean-dashboard/src/main/js/components/RunDetailsArtifacts.jsx

```diff
@@ -2,21 +2,77 @@ import React, { Component, PropTypes } from 'react';
 import { EmptyStateView, FileSize, Table } from '@jenkins-cd/design-language';
 import { Icon } from 'react-material-icons-blue';
 import Markdown from 'react-remarkable';
+import { observer } from 'mobx-react';
+import mobxUtils from 'mobx-utils';
+import { UrlConfig } from '@jenkins-cd/blueocean-core-js';
 
+const { func, object, string } = PropTypes;
+
+const ZipFileDownload = (props) => {
+    const { zipFile, t } = props;
+    if (!zipFile) {
+        return null;
+    }
+
+    return (<div className="downloadAllArtifactsButton"><a className="btn-secondary" target="_blank" title={t('rundetail.artifacts.button.downloadAll', { defaultValue: 'Download all artifact as zip' })} href={`${UrlConfig.getJenkinsRootURL()}${zipFile}`}>
+                Download All
+            </a></div>
+    );
+};
+
+ZipFileDownload.propTypes = {
+    zipFile: string,
+    t: func,
+};
+
+
+const ArtifactListingLimited = (props) => {
+    const { artifacts, t } = props;
+
+    if (!artifacts || artifacts.length < 100) {
+        return null;
+    }
+
+    return (<div className="artifactListingLimited">
+        <EmptyStateView tightSpacing><Markdown>{t('rundetail.artifacts.limit', { defaultValue: 'Only showing the first 100 artifacts' })}</Markdown></EmptyStateView>
+        </div>);
+};
+
+
+ArtifactListingLimited.propTypes = {
+    artifacts: object,
+    t: func,
+};
 /**
  * Displays a list of artifacts from the supplied build run property.
  */
+@observer
 export default class RunDetailsArtifacts extends Component {
+    componentWillMount() {
+        const { result } = this.props;
+        if (result) {
+            this.artifacts = this.context.activityService.fetchArtifacts(result._links.self.href);
```

Comment:
agree, with mobx, could this just call: `activityService.getArtifacts(activity)` during `render` where that method returns an empty list, and subsequently calls to fetch it, resulting in a re-render once it's no longer pending, so we don't need to do all this ridiculousness with `componentWillMount`?